specifyweb/frontend/js_src/lib/components/WorkBench/WbView.tsx

```diff
@@ -12,821 +12,329 @@
-        } else {
-          this.$('.wb-validate, .wb-data-check').prop('disabled', false);
-          this.$('.wb-show-upload-view')
-            .prop('disabled', false)
-            .prop('title', undefined);
-        }
-
-        /*
-         * These methods update HOT's cells settings, which resets meta data
-         * Thus, need to run them first
-         */
-        configureHandsontable(
-          this.hot!,
-          this.mappings,
-          this.dataset,
-          pickLists
-        );
-
-        if (this.dataset.rowresults) this.validation.getValidationResults();
-
-        // CHeck if any column is reordered
-        if (this.dataset.visualorder?.some((column, index) => column !== index))
-          this.hot?.updateSettings({
-            manualColumnMove: writable(this.dataset.visualorder),
-          });
-
-        this.wbUtils.findLocalityColumns();
-        this.wbUtils.render();
-
-        this.hotIsReady = true;
-
-        this.hotCommentsContainer = (document.getElementsByClassName(
-          'htComments'
-        )[0] ?? undefined) as HTMLElement | undefined;
-      });
+        : dataset.rows,
+    [dataset]
+  );
+
+  const [hotTable, setHotTable] = React.useState<HotTable | null>(null);
+  const hot = hotTable?.hotInstance ?? undefined;
+
+  const isUploaded = React.useMemo<boolean>(
+    () => dataset.uploadresult !== null && dataset.uploadresult.success,
+    [dataset]
+  );
+
+  const mappings = React.useMemo(
+    (): WbMapping | undefined => parseWbMappings(dataset),
+    [dataset]
+  );
+
+  const throttleRate = Math.ceil(clamp(10, data.length / 10, 2000));
+
+  const [hasUnsavedChanges, spreadsheetChanged, spreadsheetUpToDate] =
+    useBooleanState();
+
+  const [cellCounts, setCellCounts] = React.useState<WbCellCounts>({
+    newCells: 0,
+    invalidCells: 0,
+    searchResults: 0,
+    modifiedCells: 0,
+  });
+
+  const workbench = React.useMemo<Workbench>(() => {
+    const workbench: Workbench = {
+      data,
+      dataset,
+      hot: hot,
+      mappings: mappings!,
+      throttleRate,
+      cells: undefined!,
+      disambiguation: undefined!,
+      validation: undefined!,
+      utils: undefined!,
+      cellCounts: [cellCounts, setCellCounts],
+      spreadsheetContainerRef,
+      undoRedoIsHandled: false,
+      spreadsheetChanged,
     };
-
-    legacyLoadingContext(
-      this.initHot().then(initDataModelIntegration).catch(crash)
+    workbench.cells = new WbCellMeta(workbench);
+    workbench.disambiguation = new Disambiguation(workbench);
+    workbench.validation = new WbValidation(workbench);
+    workbench.utils = new WbUtils(workbench);
+    return workbench;
+  }, [dataset, hot]);
+
+  const checkDeletedFail = React.useCallback((statusCode: number): boolean => {
+    if (statusCode === Http.NOT_FOUND) handleDatasetDeleted();
+    return statusCode === Http.NOT_FOUND;
+  }, []);
+
+  const isMapped = Boolean(dataset.uploadplan);
```

Comment:
(nitpick) could you check for `mappings` instead of `dataset.uploadplan` here, so that `mappings` becomes the only source of truth on whether dataset is defined (since we are using `mappings` to check for that in most places rather than `dataset.uploadplan`)

the presence/absence of `dataset.uploadplan`, should be somewhat of an implementation detail, and abstracted away from all components that don't need to know the upload plan itself - instead, there is `mappings` object that also happens to provide current mappings parsed into a varierity of different useful representations